public/app/features/dashboard/components/PanelEditor/getVizualizationOptions.tsx

```diff
@@ -120,3 +99,45 @@ export function getVizualizationOptions(props: OptionPaneRenderProps): OptionsPa
 
   return Object.values(categoryIndex);
 }
+
+/**
+ * This will iterate all options panes and add register them with the configured categories
+ *
+ * @internal
+ */
+export function fillOptionsPaneItems(
+  optionEditors: PanelOptionsEditorItem[],
+  getOptionsPaneCategory: categoryGetter,
+  onValueChanged: (path: string, value: any) => void,
+  context: StandardEditorContext<any>
+) {
+  for (const pluginOption of optionEditors) {
+    if (pluginOption.showIf && !pluginOption.showIf(context.options, context.data)) {
+      continue;
+    }
+
+    const category = getOptionsPaneCategory(pluginOption.category);
+    const Editor = pluginOption.editor;
+
+    // TODO? can some options recursivly call: fillOptionsPaneItems?
```

Comment:
@torkelo -- I refactored so LayerEditor uses the same function.   We *could* explore getting nested properties from the plugin.  For example maybe:
```
pluginOption.getNestedOptions(context.options, context.data) => PanelOptionsEditorItem[]
```
and then we could recursively call `fillOptionsPaneItems`?

I think it could work -- but should be pursued in its own issue